frontend/packages/knative-plugin/src/catalog/useEventTypeProvider.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import { TFunction } from 'i18next';
+import { useTranslation } from 'react-i18next';
+import { CatalogItem, ExtensionHook, useAccessReview } from '@console/dynamic-plugin-sdk';
+import { K8sResourceKind, K8sResourceCommon } from '@console/internal/module/k8s';
+import EventType from '../components/eventing/EventType';
+import {
+  EVENT_TYPE_NAME_PARAM,
+  EVENT_TYPE_NAMESPACE_PARAM,
+  SUBSCRIBE_PROVIDER_API_VERSION_PARAM,
+  SUBSCRIBE_PROVIDER_KIND_PARAM,
+  SUBSCRIBE_PROVIDER_NAME_PARAM,
+} from '../const';
+import { useEventTypesData } from '../hooks/useEventTypesData';
+import { EventingEventTypeModel, EventingBrokerModel } from '../models';
+import { getEventSourceIcon } from '../utils/get-knative-icon';
+
+const normalizeEventType = (eventType: K8sResourceKind, t: TFunction): CatalogItem => {
+  const { kind } = EventingEventTypeModel;
+  const iconUrl = getEventSourceIcon(kind) as string;
+
+  const uid = `${eventType.metadata.namespace}-${eventType.metadata.name}`;
+
+  let provider: K8sResourceCommon = { metadata: {} };
+
+  if (eventType.spec.hasOwnProperty('reference')) {
+    provider.apiVersion = eventType.spec.reference.apiVersion;
+    provider.kind = eventType.spec.reference.kind;
+
+    if (eventType.spec.reference.hasOwnProperty('namespace')) {
+      provider.metadata.namespace = eventType.spec.reference.namespace;
+    } else {
+      provider.metadata.namespace = eventType.metadata.namespace;
+    }
+
+    if (eventType.spec.reference.hasOwnProperty('name')) {
+      provider.metadata.name = eventType.spec.reference.name;
+    }
+  } else if (eventType.spec.hasOwnProperty('broker')) {
+    provider = {
+      apiVersion: `${EventingBrokerModel.apiGroup}/${EventingBrokerModel.apiVersion}`,
+      kind: EventingBrokerModel.kind,
+      metadata: {
+        namespace: eventType.metadata.namespace,
+        name: eventType.spec.broker,
+      },
+    };
+  }
+
+  const params = new URLSearchParams();
+  params.append(SUBSCRIBE_PROVIDER_API_VERSION_PARAM, provider.apiVersion);
+  params.append(SUBSCRIBE_PROVIDER_KIND_PARAM, provider.kind);
+  params.append(SUBSCRIBE_PROVIDER_NAME_PARAM, provider.metadata.name);
+  params.append(EVENT_TYPE_NAMESPACE_PARAM, eventType.metadata.namespace);
+  params.append(EVENT_TYPE_NAME_PARAM, eventType.metadata.name);
+
+  const href = `/catalog/ns/${provider.metadata.namespace}/subscribe?${params.toString()}`;
+
+  return {
+    uid,
+    /* Add type and provider so that users can filter on event type, and provider name and namespace */
+    name: `${eventType.spec.type} (${provider.metadata.namespace}/${provider.metadata.name})`,
+    description: eventType.spec.description,
+    cta: { label: t('knative-plugin~Subscribe'), href },
```

Comment:
when clicking on subscribe what's the shown page?

Reply:
It will show a Subscribe form.

Reply:
https://github.com/user-attachments/assets/d81d54a1-07d7-478f-9d2c-73d0ec48074a

On Click of Subscribe I am getting page not found error.

URL trying to get redirect to: http://localhost:9000/catalog/ns/prabhu/subscribe?providerApiVersion=eventing.knative.dev%2Fv1&providerKind=Broker&providerName=my-broker&eventTypeNamespace=prabhu&eventTypeName=et-my-broker-30a073990e787ff3496813edcc5fdc1a

Reply:
Yes, as the Subscribe form will be covered in this story https://issues.redhat.com/browse/ODC-7664 as another PR.  This PR includes only a Catalog for the Knative Events.

Reply:
Thanks, makes sense splitting for easier reviews, the rest looks good to me